src/psyclone/tests/domain/common/algorithm/alg_psyir_test.py

```diff
@@ -370,6 +380,27 @@ def test_aic_createpsylayersymbolrootnames():
     assert "No Routine or Container node found." in str(error.value)
 
 
+def test_aic_defcontainerrootname():
+    '''Check that _def_container_root_name returns the expected value'''
+    code = (
+        "subroutine alg1()\n"
+        "  use kern_mod, only : kern\n"
+        "  use field_mod, only : field_type\n"
+        "  type(field_type) :: field1\n"
+        "  call invoke(kern(field1))\n"
+        "end subroutine alg1\n")
+    psyir = create_alg_psyir(code)
+    invoke = psyir.children[0][0]
+    assert isinstance(invoke, AlgorithmInvokeCall)
+    result_node = None
+    for node in psyir.root.walk((Routine, Container)):
```

Comment:
This looks overly complicated given that the code below only seems to expect a single value for `result_node` and this test only works with a single input code fragment.

Reply:
Good point. I just copied the generic logic without thinking about it. I've now simplified for this particular example.